src/main/java/com/teragrep/pth_10/executor/DPLExecutorImpl.java

```diff
@@ -97,149 +98,167 @@ private DPLAuditInformation setupAuditInformation(String query) {
-        catalystContext.setBaseUrl(config.getString("dpl.web.url"));
-        LOGGER.debug("Setting notebook url");
-        catalystContext.setNotebookUrl(noteId);
-        LOGGER.debug("Setting paragraph url");
-        catalystContext.setParagraphUrl(paragraphId);
-
-        LOGGER.debug("Creating lexer");
-        DPLLexer lexer = new DPLLexer(CharStreams.fromString(lines));
-        // Catch also lexer-errors i.e. missing '"'-chars and so on. 
-        lexer.addErrorListener(new DPLErrorListenerImpl("Lexer"));
-
-        LOGGER.debug("Creating parser");
-        DPLParser parser = new DPLParser(new CommonTokenStream(lexer));
-        LOGGER.debug("Setting earliest");
-        catalystContext.setEarliest("-1Y"); // TODO take from TimeSet
-        LOGGER.debug("Creating visitor");
-        DPLParserCatalystVisitor visitor = new DPLParserCatalystVisitor(catalystContext);
-
-        // Get syntax errors and throw then to zeppelin before executing stream handling
-        LOGGER.debug("Added error listener");
-        parser.addErrorListener(new DPLErrorListenerImpl("Parser"));
-
-        ParseTree tree;
         try {
-            LOGGER.debug("Running parser tree root");
-            tree = parser.root();
-        }
-        catch (IllegalStateException e) {
-            return new DPLExecutorResultImpl(DPLExecutorResult.Code.ERROR, e.toString());
-        }
-        catch (StringIndexOutOfBoundsException e) {
-            final String msg = "Parsing error: String index out of bounds. Check for unbalanced quotes - "
-                    + "make sure each quote (\") has a pair!";
-            return new DPLExecutorResultImpl(DPLExecutorResult.Code.ERROR, msg);
-        }
-
-        // set output consumer
-        LOGGER.debug("Creating consumer");
-        visitor.setConsumer(batchHandler);
-
-        // set BatchCollect size
-        LOGGER.debug("Setting recall size");
-        visitor.setDPLRecallSize(config.getInt("dpl.recall-size"));
-
-        LOGGER.debug("Creating translationResultNode");
-        TranslationResultNode n = (TranslationResultNode) visitor.visit(tree);
-        DataStreamWriter<Row> dsw;
-        if (n == null) {
-            return new DPLExecutorResultImpl(
-                    DPLExecutorResult.Code.ERROR,
-                    "parser can't construct processing pipeline"
-            );
-        }
-        // execute steplist
-        try {
-            dsw = n.stepList.execute();
-        }
-        catch (Exception e) {
-            // This will also catch AnalysisExceptions, however Spark does not differentiate between
-            // different types, they're all Exceptions.
-            // log initial exception
-            LOGGER.error("Got exception: <{}>:", e.getMessage(), e);
-
-            // get root cause of the exception
-            Throwable exception = e;
-            while (exception.getCause() != null) {
-                exception = exception.getCause();
+            MDC.put("queryName", queryName);
```

Comment:
is mdc proper way here? we need the same information quite deep in the stack. i.e. pth_06 logs need to be able to produce the queryId too..